### PR TITLE
[test]: Add BGP GR helper mode test.

### PR DESCRIPTION
**Pull request by oleksandrivantsiv**
- Add BGP GR helper mode test.
- Simplify init/analyze procedure for loganalyzer.

### CODE REVIEW

File: ansible/roles/test/tasks/bgp_gr_helper.yml [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+      bgp_gr_stalepath_timer: "{{ (bgp_gr_timer|int + 30) }}"
+
+- name: Set bgp stalepath timer.
+  command: vtysh -c "conf t" -c "router bgp {{ minigraph_bgp_asn }}" -c "bgp graceful-restart stalepath-time {{ bgp_gr_stalepath_timer }}"
+
+# When RIBD up and send bgp open message it will set F bit to 1. Which means that during restart
+# all routes were preserved in FIB. When DUT receives open message with F bit set to 1 it also
+# should preserve all routes (no route update should happens).
+- name: Force stop RIBD to simulate GR.
+  shell: "killall -9 ribd; sleep 0.5; ifconfig et1 down"
```
Comment (lguohan) [inline ansible/roles/test/tasks/bgp_gr_helper.yml]: can you describe the ribd behavior after you kill? is it going to restart immediately? why do you need to shutdown et1? Add them to the comments?

File: ansible/roles/test/tasks/bgp_gr_helper.yml [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+  delegate_to: "{{ vm_ip }}"
```
Comment (lguohan) [inline ansible/roles/test/tasks/bgp_gr_helper.yml]: how can you login the bgp vm? using what account? do we setup those account in the vm config template?

File: ansible/roles/test/tasks/bgp_gr_helper.yml [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+  vars:
+    test_match_file: routes_update_match.txt
+
+- include: roles/test/files/tools/loganalyzer/loganalyzer_init.yml
+  vars:
+    test_expect_file: routes_update_expect.txt
+
+- name: Restart the VM
+  shell: killall -9 ribd ; reboot
+  delegate_to: "{{ vm_ip }}"
```
Comment (lguohan) [inline ansible/roles/test/tasks/bgp_gr_helper.yml]: can you explain why kill ribd before reboot?

File: ansible/roles/test/tasks/bgp_gr_helper.yml [hunk 4]
```diff
@@ -0,0 +1,102 @@
+#-----------------------------------------
+# Run BGP GR helper mode test and Perform log analysis.
+#-----------------------------------------
```
Comment (lguohan) [inline ansible/roles/test/tasks/bgp_gr_helper.yml]: this test needs description on how it works.

File: ansible/roles/test/files/tools/loganalyzer/loganalyzer_analyze.yml [hunk 5]
```diff
@@ -4,7 +4,28 @@
 
 - debug: msg="starting loganalyzer analysis phase"
 
-- set_fact: cmd="python {{ run_dir }}/loganalyzer.py --action analyze --run_id {{ testname_unique }} --out_dir {{ test_out_dir }} -m {{ match_file }} -i {{ ignore_file }} -e {{ test_expect_file }} -v"
+- name: Init variables
```
Comment (pavel-shirshov) [inline ansible/roles/test/files/tools/loganalyzer/loganalyzer_analyze.yml]: Can you please create a separate PR with loganalyzer changes?

File: ansible/roles/test/tasks/bgp_gr_helper.yml (hunk 2 above)
Comment (oleksandrivantsiv) [inline ansible/roles/test/tasks/bgp_gr_helper.yml]: All information about VM is initialized in get_vm_info.yml.
VM IP address and account are taken using lldp.yml. It gathers information about VMs and puts it into in memory inventory.

**Comment by oleksandrivantsiv** [comment]
@pavel-shirshov Loganalizer changes were moved to https://github.com/Azure/sonic-mgmt/pull/230

**Comment by oleksandrivantsiv** [comment]
This PR depends on https://github.com/Azure/sonic-mgmt/pull/230

**Comment by oleksandrivantsiv** [comment]
@lguohan Added comments. Please review

**Comment by lguohan** [comment]
can you cherry-pick the commit given the history is messed up.

**Comment by oleksandrivantsiv** [comment]
@lguohan done

File: ansible/roles/eos/templates/t1-lag-spine.j2 [hunk 6]
```diff
@@ -97,6 +97,9 @@ route-map PREPENDAS permit {{ 2 * (podset * props.tor_number + tor + 1) + 1 }}
 router bgp {{ host['bgp']['asn'] }}
  router-id {{ host['interfaces']['Loopback0']['ipv4'] |  ipaddr('address') }}
  !
+ graceful-restart restart-time {{ bgp_gr_timer }}
+ graceful-restart
+ !
```
Comment (pavel-shirshov) [inline ansible/roles/eos/templates/t1-lag-spine.j2]: Is the possible to insert the changes before the test and remove them after the test?

File: ansible/roles/eos/templates/t1-lag-tor.j2 [hunk 7]
```diff
@@ -59,6 +59,9 @@ interface {{ name }}
 router bgp {{ host['bgp']['asn'] }}
  router-id {{ host['interfaces']['Loopback0']['ipv4'] |  ipaddr('address') }}
  !
+ graceful-restart restart-time {{ bgp_gr_timer }}
+ graceful-restart
+ !
```
Comment (pavel-shirshov) [inline ansible/roles/eos/templates/t1-lag-tor.j2]: The same as previous

File: ansible/roles/eos/templates/t1-spine.j2 [hunk 8]
```diff
@@ -90,6 +90,9 @@ route-map PREPENDAS permit {{ 2 * (podset * props.tor_number + tor + 1) + 1 }}
 router bgp {{ host['bgp']['asn'] }}
  router-id {{ host['interfaces']['Loopback0']['ipv4'] |  ipaddr('address') }}
  !
+ graceful-restart restart-time {{ bgp_gr_timer }}
+ graceful-restart
+ !
```
Comment (pavel-shirshov) [inline ansible/roles/eos/templates/t1-spine.j2]: The same as previous

File: ansible/roles/eos/templates/t1-tor.j2 [hunk 9]
```diff
@@ -59,6 +59,9 @@ interface {{ name }}
 router bgp {{ host['bgp']['asn'] }}
  router-id {{ host['interfaces']['Loopback0']['ipv4'] |  ipaddr('address') }}
  !
+ graceful-restart restart-time {{ bgp_gr_timer }}
+ graceful-restart
+ !
```
Comment (pavel-shirshov) [inline ansible/roles/eos/templates/t1-tor.j2]: The same as previous

**Pull request closed without merging**